test/generic/MPoly-test.jl

```diff
@@ -113,18 +113,55 @@
    @test y isa Generic.MPoly{Generic.Poly{BigInt}}
    @test z isa Generic.MPoly{Generic.Poly{BigInt}}
 
-   ZZxyz, (x,y,z) = polynomial_ring(ZZ, 'x':'z')
-   @test ZZxyz isa Generic.MPolyRing
-
-   ZZxyz2, (x2,y2,z2) = polynomial_ring(ZZ, (:x, 'y', GenericString("z")))
-   @test ZZxyz == ZZxyz2
-   @test (x,y,z) == (x2,y2,z2)
-
-   ZZxyz3, _ = polynomial_ring(ZZ, Union{String,Char,Symbol}["x", 'y', :z])
-   @test ZZxyz == ZZxyz3
-
-   ZZxyz4, _ = ZZ["x", 'y', :z]
-   @test ZZxyz == ZZxyz4
+   ZZxyz_ = polynomial_ring(ZZ, 'x':'z')
+   ZZxyz2, xyz2... = polynomial_ring(ZZ, (:x, 'y', GenericString("z")))
+   ZZxyz3, xyz3... = polynomial_ring(ZZ, :x, 'y', GenericString("z"))
+   ZZxyz4_ = polynomial_ring(ZZ, Union{String,Char,Symbol}["x", 'y', :z])
+   ZZxyz5_ = ZZ["x", 'y', :z]
+   ZZxyz6 = @polynomial_ring ZZ x y z
+
+   @test ZZxyz_[1] isa Generic.MPolyRing
+   @test ZZxyz_ == (ZZxyz2, collect(xyz2))
+   @test ZZxyz_ == (ZZxyz3, collect(xyz3))
+   @test ZZxyz_ == ZZxyz4_
+   @test ZZxyz_ == ZZxyz5_
+   @test ZZxyz_ == (ZZxyz6, [x, y, z])
+
+   ZZxxx0_ = polynomial_ring(ZZ, :x=>Base.OneTo(3))
+   ZZxxx_ = polynomial_ring(ZZ, :x=>1:3)
+   ZZxxx2, (xxx2,) = ZZ[:x=>1:3]
+   ZZxxx3_ = polynomial_ring(ZZ, "x[@]"=>1:3)
+
+   @test ZZxxx_[1] isa Generic.MPolyRing
+   @test ZZxxx_ == ZZxxx0_
+   @test ZZxxx_ == (ZZxxx2, xxx2)
+
+   QQxxx_ = polynomial_ring(QQ, "x#" => 1:3)
+   QQxxx2 = @polynomial_ring QQ x[1:3]
+
+   @test QQxxx_[1] isa Generic.MPolyRing
+   @test QQxxx_ == (QQxxx2, [x1, x2, x3])
+
+   ZZxy_ = polynomial_ring(ZZ, :x => (1:2, 1:2), :y => 0:3)
+   ZZxy2_ = polynomial_ring(ZZ, :x => ["1,1" "1,2"; "2,1" "2,2"], :y => (0:3,))
+   ZZxy3_ = polynomial_ring(ZZ, "x[@]" => ["1,1" "1,2"; "2,1" "2,2"], "y" => 0:3)
+   ZZxy4_ = polynomial_ring(ZZ, "x[@,@]" => (1:2, 1:2), "y[#]" => 0:3)
+   ZZxy5_ = polynomial_ring(ZZ, "x[%s,%s]" => (1:2, 1:2), "y[%s]" => 0:3)
+
+   @test ZZxy_[1] isa Generic.MPolyRing
+   @test ZZxy_ == ZZxy2_
+   @test ZZxy_ == ZZxy3_
+   @test ZZxy_ == ZZxy4_
+   @test ZZxy_ == ZZxy5_
+
+   QQxy_ = polynomial_ring(QQ, "x#" => (1:2, 1:2), Symbol.(:y, 0:3))
+   QQxy2 = @polynomial_ring QQ x[1:2,1:2] y[0:3]
```

Comment:
Ahh, I think we had a misunderstanding here still: I would much rather prefer we don't introduce *yet another syntax* for `@polynomial_ring` versus `polynomial_ring`.

In other words, my intention was that `@polynomial_ring` is used exactly like `polynomial_ring` (including parenthesis etc.), the only difference is that it returns a single object (the new ring), and auto-assign in the current scope variables with suitable names (or raises an error if the names are not valid identifiers). So:
```
R = @polynomial_ring(QQ, [:x, :y])    # assign Julia variables x and y
R = @polynomial_ring(QQ, "x#" => 1:3) # assign Julia variables x1, x2, x3
R = @polynomial_ring(QQ, "x" => 1:3)  # also assign x1, x2, x3
R = @polynomial_ring(QQ, "x" => -1:3) # error, "x-1" not a valid identifier
R = @polynomial_ring(QQ, "x" => 0:10) # works, assigns x0 ... x10
R = @polynomial_ring(QQ, "x" => 1:n)  # works in REPL, probably raises an error in package code, unless `n` is a global constant
R = @polynomial_ring(QQ, "x" => (0:10,0:10) # works, but does silly things, e.g. defines x010 and x1010
R = @polynomial_ring(QQ, "x" => (0:100,0:100) # error because x111 is ambiguous
            # could be detected by general code which ensures all variable names are unique
```


Rationale: Yes, it might "look nicer" but as long as it can only be used in `@polynomial_ring` but not `polynomial_ring` it is bad from a "learning how to use OSCAR" perspective: you start by writing REPL code, and of course use `@polynomial_ring` ... and when you later have to write real functions, you need to learn how to use `polynomial_ring` all over. And of course when switching between the two, people will invariably mix up which syntax to use where..

Reply:
Okay. We can `eval`uate the entire variable name specification.

Reply:
While this design will purge some code, it will also create new trouble when calling macros with keyword arguments (as for weights).

Reply:
Yup, that was my thought exactly.

Reply:
I assume you meant `@polynomial_ring(QQ, "x" => 1:3)` ? I am torn on that one. On the one hand, there is consistency and the principle of least surprise, which would suggest to use `x[1]` etc. On the other hand, users like @wdecker will probably prefer if they get `x1`... Question is how "hard" it will be for them to accept / remember to use `polynomial_ring(QQ, "x#" => 1:3)`.

Feel free to implement whatever is easier for you right now (probably just using the same as without `@`, i.e., `x[1]`), and we can revise this later if and as needed.

Reply:
Alright. I have now implemented this and went with "x[1]" indeed.